docs/source/miscellaneous/doc/adr/0013-write-cassandra-tombstones-and-sentinels-with-a-fresh-cassandra-timestamp.md

```diff
@@ -0,0 +1,109 @@
+the tombstone was covering that now becomes readable).
+
+Since Atlas writes its sweep sentinels at timestamp `-1`, some SSTables will have a minimum timestamp of `-1` which
+may prevent compactions on subsets of SSTables from dropping many legitimately droppable tombstones.
+
+## Decision
+
+Change the timestamps at which sweep sentinels, tombstones and range tombstones are written to be a fresh timestamp
+from the timestamp service. In the case of sentinels and tombstones, a single API call to add a set of garbage
+collection sentinels or delete a set of cells makes one call to the fresh timestamp service. Cell write timestamps
+are unchanged (so a write at timestamp `TS` still receives a Cassandra timestamp of `TS`).
+
+The write pattern for the metadata table is unchanged (still uses wall-clock time). Some of this is because of a
+chicken and egg problem with embedded users of the timestamp service; additionally, there are existing CLIs that
+manipulate the metadata table but don't have ready access to a timestamp service.
+
+### Proof of Correctness/Safety
+Recall the invariants we seek to preserve:
+
+1. a tombstone or range tombstone that covers a cell with the Cassandra timestamp `TS` must have a Cassandra timestamp 
+  greater than `TS`
+2. a tombstone or range tombstone that covers a sweep sentinel with Cassandra timestamp `TS` must have a Cassandra
+  timestamp greater than `TS`
+3. the insertion of a fresh sweep sentinel after a tombstone or range tombstone at time `TS` must have a Cassandra
+  timestamp greater than `TS`
+
+A cell being covered with Cassandra timestamp `TS` must also have Atlas start timestamp `TS`. Given that this cell is
+already written to the database, a fresh timestamp `TS'` is necessarily greater than `TS`, giving us statement 1.
+
+Statements 2 and 3 follow from the guarantees of the timestamp service.
+
+### Migrations/Cutover
+Notice that even in the presence of values written under the old scheme, our invariants are preserved.
+Statement 1 holds because cell write timestamps haven't changed (and we can re-use the proof above).
+For statements 2 and 3, the timestamps of values we are covering would be `-1` and `deletionTimestamp + 1`;
+we know that a fresh timestamp will be at least that (since timestamps aren't given out twice).
+
+Timestamps for existing values are not changed, so a major compaction may be necessary before one begins to 
+reap the benefits of better tombstone droppability.
+
+## Consequences
+
+- Key-value services now need a fresh timestamp supplier for initialisation.
+  - An alternative approach where this was optional was considered (and if not provided we would use the legacy mode).
+    We decided to mandate providing the supplier as the benefits of better tombstone droppability are significant.
+  - The fresh timestamp supplier is unused by key value services other than Cassandra.
+- Inserting sentinels or writing tombstones/range tombstones requires one additional RPC.
```

Comment:
Same comment I made above about sentinels and deletes not being critical path is great; critical path reads/writes aren't affected by the extra rpc call

Reply:
Yep, added one point underneath this one saying that.